packages/rum/src/domain/record/record.ts

```diff
@@ -57,6 +58,24 @@ export function record(options: RecordOptions): RecordAPI {
     })
   }
 
+  if (isExperimentalFeatureEnabled('visualviewport')) {
+    if (visualViewport) {
+      const viewport = visualViewport
+      emit({
+        data: {
+          scale: viewport.scale,
+          offsetLeft: viewport.offsetLeft,
+          offsetTop: viewport.offsetTop,
+          pageLeft: viewport.pageLeft,
+          pageTop: viewport.pageTop,
+          height: viewport.height,
+          width: viewport.width,
+        },
+        type: RecordType.VisualViewport,
```

Comment:
**question:** is there a web-ui PR sister PR to handle this new type of record?

If not, it should not break the player but this records won't have any handler and therefore will trigger an error in error tracking.
That said, to have the new types in web-ui, you'll need to publish this on NPM

Reply:
Web-UI has a corresponding listener here, though not ready for review:
```
 recordsEmitter.addEventListener(
      RecordType.VisualViewportRecord,
      visualViewportCallback
 );
```

Reply:
@BenoitZugmeyer agreed